src/EditModel/event-program/tracker-program/assign-tracked-entity-attributes/AssignAttributes.js

```diff
@@ -152,17 +98,26 @@ function AssignAttributes(props, { d2 }) {
 
     // Assign existing attributes
     assignedItemStore.setState(
-        props.items.map(a => a.trackedEntityAttribute.id),
+        props.assignedAttributes.map(a => a.trackedEntityAttribute.id),
     );
 
+    const onMoveAttributes = (newAttributesOrderIds) => {
+        assignedItemStore.setState(newAttributesOrderIds);
+        // need to update this.props.assignedAttributes to reflect new order in epics
+    };
+
     // Create edit-able rows for assigned attributes
-    const tableRows = props.items
+    const tableRows = props.assignedAttributes
         .map(addDisplayProperties(props.availableAttributes))
-        .map((programAttribute, index) => (
-            <PureProgramAttribute
+        .map(programAttribute => (
+            <ProgramAttributeRow
                 key={programAttribute.id}
-                programAttribute={programAttribute}
-                onEditProgramAttribute={props.onEditProgramAttribute}
+                displayName={programAttribute.trackedEntityAttribute.displayName}
+                attribute={programAttribute}
+                onEditAttribute={props.onEditProgramAttribute}
+                isDateValue={programAttribute.trackedEntityAttribute.valueType === 'DATE'}
+                isUnique={programAttribute.trackedEntityAttribute.unique}
+                hasOptionSet={!!programAttribute.trackedEntityAttribute.optionSet}
```

Comment:
is the `ProgramAttributeRow` used elsewhere? Looks like it's reading a lot of props from the same object, which is also passed in, so looks the values could be spread on the inside of `ProgramAttributeRow` component?

Reply:
Ah, yes. It also used in trackedEntityType. Spreading would make it non-generic.

Reply:
Ok.